src/pilot/assign_role.py

```diff
@@ -321,10 +321,10 @@ def define_compute_logical_disks(drac_client, raid_controller_name):
     raid_10_logical_disk = define_single_raid_10_logical_disk(
         drac_client, raid_controller_name)
 
-    if raid_10_logical_disk is None:
-        return None
-
     logical_disks = list()
+    if not raid_10_logical_disk:
```

Comment:
I don't think this is right.  define_single_raid_10_logical_disk can return either an empty dict or None.  None is meant to indicate that an error occurred, which should cause assign_role to exit.  The way this code is written, dict() and None will enter this "if" clause and return an empty list in both of those scenarios.  This will cause define_target_raid_config to return a dictionary in both cases, and that will cause this "if" statement to be entered (https://github.com/dsp-jetpack/JetPack/pull/205/files#diff-a1208342b5ece66e825ba657f52e8a16R783), and assign_role will continue along even in the error case.
I think this code should return a None if a None is returned from define_single_raid_10_logical_disk, and an empty list if an empty dict is returned, instead of returning an empty list in both cases.  I believe that would cause assign_role to abort in the event that an error occurs resulting in None being returned.

Reply:
As you suggested, I have modified the changes on current PR. Please review the same.